src/app/src/main/kotlin/org/vaccineimpact/api/app/controllers/NewStyleOneTimeLinkController.kt

```diff
@@ -0,0 +1,50 @@
+package org.vaccineimpact.api.app.controllers
+
+import org.vaccineimpact.api.app.RedirectValidator
+import org.vaccineimpact.api.app.app_start.Controller
+import org.vaccineimpact.api.app.context.ActionContext
+import org.vaccineimpact.api.app.repositories.Repositories
+import org.vaccineimpact.api.app.repositories.TokenRepository
+import org.vaccineimpact.api.models.helpers.OneTimeAction
+import org.vaccineimpact.api.security.WebTokenHelper
+import java.time.Duration
+
+class NewStyleOneTimeLinkController(
+        context: ActionContext,
+        private val repo: TokenRepository,
+        private val tokenHelper: WebTokenHelper,
+        private val redirectValidator: RedirectValidator = RedirectValidator())
+    : Controller(context)
+{
+    constructor(context: ActionContext, repositories: Repositories, tokenHelper: WebTokenHelper)
+            : this(context, repositories.token, tokenHelper)
```

Comment:
No other controller is going to use the `tokenHelper`, so I'm a little reluctant for it to be a required controller parameter. (It feels ok for `Repositories` to be required because every controller needs at least one repo, and it's functioning as a dependency container to inject the right dependencies to the primary constructors.)

`KeyHelper` stores the key pair and is a singleton, so actually we're free to instantiate `WebTokenHelper` whenever right? Meaning we could pass a new instance of it to this controller without it needing to be in the base controller constructor. What do you think?

Reply:
:+1: